### ISSUE TITLE
unable to merge ways that belong to different relations

### ISSUE THREAD

**Issue by mertless** [openstreetmap/iD#8887]
### How to reproduce the issue?

when I try to merge ways for roads, an error saying ways have conflicting relations appears and greys out the merge button, even if the other way has no relations.

### Screenshot(s) or anything else?

![Screenshot (2)](https://user-images.githubusercontent.com/97324182/148630918-6427e538-1d6e-42d3-bfde-ee64220dcdc0.png)
![Screenshot (3)](https://user-images.githubusercontent.com/97324182/148630933-578beacd-cc90-44ac-8506-721250b1b299.png)

### Which iD Editor versions do you see the issue on?

Released version at openstreetmap.org/edit

### Which browsers are you seeing this problem on?

Chrome

**Comment by 1ec5** [comment]
I think this is a duplicate of #8779. The eventual fix might need different logic for routes versus multipolygons, but the root cause is the same.

**Comment by boothym** [comment]
This is a bit annoying as the merging behaviour is different to when you merge nodes/way which have identical tags - maybe a prompt of some sort could ask if you really want to add the way to the relation as well as merging?

**Comment by blackboxlogic** [comment]
I'm getting this same error. I tried manually adding the relations to the second way, but I didn't add the relations in the same _ORDER_ so it still wasn't good enough. I removed the relations and added them in same order as the first way, and then it allowed the merge.

Why/when/how could the order of relations on an element possibly matter?

When I merge ways, I intend to union thier relations. At most, give me a confirmation dialog with a warning.

**Comment by matkoniecz** [comment]
Is it still happening? If yes, can you give example of affected ways?

**Comment by boothym** [comment]
Yes I've just tested merging a line without a relation to a line with a relation, and they do not merge.

**Comment by matkoniecz** [comment]
Oh, then title of issue should be edited to be more clear

**Comment by matkoniecz** [comment]
Note that at least in some cases it will be validly blocked. In which cases not blocking would be desirable?

Are there any cases where it should NOT be blocked?

**Comment by tyrasd** [comment]
> when I try to merge ways for roads, an error saying ways have conflicting relations appears

This is an intended behaviour of iD: If it were allowed to merge ways as proposed, it would lead to many (e.g. route) relations to become invalid. If you want to intentionally extend a route relation, please manually add the relations to the added way.

> Why/when/how could the order of relations on an element possibly matter?

The order does matter for route relations (e.g. bus routes). that said, I could not reproduce the reported scenario where the membership order prevented the ways from being merged.

> Are there any cases where it should NOT be blocked?

There are some cases that are already implemented, e.g. for `type=restriction`. If there are more that should be handled similarly, please open a new issue with the concrete situation and relation type.

**Comment by boothym** [comment]
> If it were allowed to merge ways as proposed, it would lead to many (e.g. route) relations to become invalid.

What do you mean by this, why would many relations become invalid?

> If you want to intentionally extend a route relation, please manually add the relations to the added way.

I understand if the ways have different tags or have different relations, they cannot be merged.

But a way with tags can be merged to a way without tags - so why can't a way with a relation be merged to a way without a relation?

The reason for the merge function is so you don't need to copy and paste or add all the tags manually, same should apply for relations.

**Comment by matkoniecz** [comment]
> What do you mean by this, why would many relations become invalid?

if you have path with route relations then merging it with another path will in nearly all cases break route relation

**Comment by matkoniecz** [comment]
> why can't a way with a relation be merged to a way without a relation?

because more often than not it would break relations

if iD blocks some cases where it would be safe, see:

> >  Are there any cases where it should NOT be blocked?
>
> There are some cases that are already implemented, e.g. for type=restriction. If there are more that should be handled similarly, please open a new issue with the concrete situation and relation type.
